feat: #1684 Ability to add custom popup switches

AlexanderArvidsson opened the pull request:
Adds the ability to add custom argument switches to any popups by allowing users to create custom builder hooks to modify popups while they're building.
![image](https://github.com/user-attachments/assets/9948f081-d3b6-40ad-8468-b6a9d0e9ab0b)

The above is implemented in config like this to provide the `-o merge_request.create` GitLab specific option:
```lua
require('neogit').setup({
  builders = {
    NeogitPushPopup = function(builder)
      --
      builder:switch('m', 'merge_request.create', 'Create merge request', { cli_prefix = '-o ', persisted = false })
    end,
  },
})
```

This PR also introduces the `persisted = false` option for switches, and it also migrates all existing non-persisted switches to this flag instead of the `ignored_settings` option. It doesn't remove `ignored_settings`, as people might have overridden it.
It also adds `persisted = false` to some extra popups that wasn't covered before (mostly `force` switches).

Fixes #1684.

## Future changes
* Modify existing switches' options, to change their keybinds, text, persisted option, etc.
* Set an ordering for switches, or the ability to _inject_ new options anywhere in the list.

CKolkey commented:
Good stuff! A quick look at the code doesn't show any glaring issues. But would you mind adding some documentation to the help file? Under API or something that makes sense :)

AlexanderArvidsson commented:
I added some docs under `neogit_custom_popups` and also updated the references to the default config, which has now changed (migrated to `persisted` flags, it _shouldn't_ introduce any breaking changes afaict.

Let me know if you want to make some changes or move it somewhere else!

CKolkey commented:
This is great, thanks a ton for implementing it!
